main/streams/memory.c

```diff
@@ -240,6 +241,7 @@ static int php_stream_memory_set_option(php_stream *stream, int option, int valu
 						size_t old_size = ZSTR_LEN(ms->data);
 						ms->data = zend_string_realloc(ms->data, newsize, 0);
 						memset(ZSTR_VAL(ms->data) + old_size, 0, newsize - old_size);
+						ZSTR_VAL(ms->data)[ZSTR_LEN(ms->data)] = '\0';
```

Comment:
This could be simplified
```suggestion
						memset(ZSTR_VAL(ms->data) + old_size, 0, newsize - old_size + 1); // +1 to zero-terminate
```